File: sparkcql-synthetic/src/main/scala/gov/va/sparkcql/synthea/SyntheaDataAdapterFactory.scala
package gov.va.sparkcql.synthea

import org.apache.spark.sql.{SparkSession, DataFrame}
import org.apache.spark.sql.functions._
import gov.va.sparkcql.di.ComponentConfiguration
import gov.va.sparkcql.adapter.data.{DataAdapter, DataAdapterFactory}

class SyntheaDataAdapterFactory() extends DataAdapterFactory {

  override val defaultConfiguration: Option[ComponentConfiguration] = {
    val configuration = new ComponentConfiguration()
    configuration.write("size", "none")
    Some(configuration)
  }

  override def create(configuration: ComponentConfiguration, spark: SparkSession): DataAdapter = {
    val size = configuration.read("sparkcql.syntheadataadapter.samplesize") match {
      case "10" => SampleSize.SampleSize10
      case "1000" => SampleSize.SampleSize1000
      case _ => SampleSize.SampleSizeNone
    }
    new SyntheaDataAdapter(spark, size)
  }
}

File: sparkcql-synthetic/src/main/java/gov/va/sparkcql/synthetic/SyntheaDataAdapterFactory.java
package gov.va.sparkcql.synthetic;

import org.apache.spark.sql.SparkSession;

import gov.va.sparkcql.Configuration;
import gov.va.sparkcql.adapter.data.DataAdapter;
import gov.va.sparkcql.adapter.data.DataAdapterFactory;

public class SyntheaDataAdapterFactory implements DataAdapterFactory {

    protected static final String CONFIG_POPULATION_SIZE = "sparkcql.synthea.size";

    @Override
    public DataAdapter create(Configuration configuration, SparkSession spark) {
        var size = PopulationSize.valueOf(configuration.read(CONFIG_POPULATION_SIZE, "PopulationSizeNone"));
        return new SyntheaSourceAdapter(spark, size);
    }
}